app/src/main/java/edu/ucsc/cmps115_spring2017/face2name/MainScreen.java

```diff
@@ -67,42 +63,73 @@ public void onPreviewReady() {
+    }
+    
     public boolean onTouchEvent(MotionEvent event) {
-        int l_x = mLayerView.getRectPoint("Left X");
-        int r_x = mLayerView.getRectPoint("Right X");
-        int u_y = mLayerView.getRectPoint("Upper Y");
-        int l_y = mLayerView.getRectPoint("Lower Y");
-
-        Log.e("PRESS", "LX: " + l_x + " RX: " + r_x + "UY: " + u_y + "LY: " + l_y);
 
+        //gets the coordinate of press event
         int touchX = (int) event.getX();
         int touchY = (int) event.getY();
-        Log.e("PRESS", "GET_X: "+touchX+ " GET_Y: "+touchY);
-        Log.e("PRESS", "EVENT ACTION: "+event.getAction()+ " ACTION BUTTON PRESS: "+ MotionEvent.ACTION_BUTTON_PRESS);
-        if (event.getAction() == MotionEvent.ACTION_BUTTON_PRESS ||
-                event.getAction() == MotionEvent.ACTION_BUTTON_RELEASE ||
-                event.getAction() == MotionEvent.ACTION_DOWN ||
-                event.getAction() == MotionEvent.ACTION_UP) {
-            Log.e("PRESS", "RECT PRESSED");
-            if ((touchX >= l_x && touchX <= r_x) && (touchY >= u_y && touchY <= l_y)) {
-                AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(
-                        this);
-
-                // set title
-                alertDialogBuilder.setTitle("Rectangle Pressed");
-
-                // create alert dialog
-                AlertDialog alertDialog = alertDialogBuilder.create();
-
-                // show it
-                alertDialog.show();
+
+
+        if (isPressEvent(event.getAction()))
+        {
+            // try/catch block because getRectPoint throws an IllegalArgumentException
+            try{
+                //returns the coordinates of the red rectangle
+                int leftXCoordinate = mLayerView.getRectPoint(LayerView.coordinate_sections.LEFT_X);
+                int rightXCoordinate = mLayerView.getRectPoint(LayerView.coordinate_sections.RIGHT_X);
+                int upperYCoordinate = mLayerView.getRectPoint(LayerView.coordinate_sections.UPPER_Y);
+                int lowerYCoordinate = mLayerView.getRectPoint(LayerView.coordinate_sections.LOWER_Y);
+
+                //If the press was within the X bounds and within the Y bounds, continue
+                if ((touchX >= leftXCoordinate && touchX <= rightXCoordinate) &&
+                        (touchY >= upperYCoordinate && touchY <= lowerYCoordinate)) {
+
+                   //build dialong
+                    AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
+
+                    // set title
+                    alertDialogBuilder.setTitle("Rectangle Pressed");
+
+                    // create alert dialog
+                    AlertDialog alertDialog = alertDialogBuilder.create();
+
+                    // show it
+                    alertDialog.show();
+                }
+            }
+            catch(IllegalArgumentException iArg){
+                    Log.d("ILLEGALARGUMENT", "IllegalArgumentException: " + iArg);
```

Comment:
It's not strictly necessary to handle the exception thrown by `getRectPoint()`. In fact, I would even recommend against it. The end user should never encounter this exception because its sole purpose is to keep the programmer in check. It ensures that every value of `enum coordinate_sections` has a case in `getRectPoint()`'s switch statement. Programmers tend to respond more promptly to a crash than a line in a debug log.